MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/MovieDetailsRepository.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  GetMovieDetails-Repository.swift
+//  MovieAppPetProject
+//
+//  Created by Sebastian Jacobs on 2024/04/08.
+//
+
+import Foundation
+
+protocol MovieDetailsRepositoryType {
+    func fetchMovieDetails(movieID: Int, completion: @escaping (Result<MovieDetails, CustomError>) -> Void)
+    func addToWatchlist(movieDetails: MovieDetails)
+}
+
+class MovieDetailsRepository: MovieDetailsRepositoryType {
+
+    // MARK: - Variable
+    private let networkManager = NetworkManager()
+    private let coreDataManager: CoreDataManager
```

Comment:
if you're injecting coredata, then inject networkManager as well

Reply:
done